modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteReduceIntersect.java

```diff
@@ -34,55 +34,68 @@
  * Physical node for REDUCE phase of INTERSECT operator.
  */
 public class IgniteReduceIntersect extends IgniteIntersect implements IgniteReduceSetOp {
+
+    private final int inputsNum;
+
     /**
      * Constructor.
-     * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
+     *
+     * @param cluster  Cluster that this relational expression belongs to.
+     * @param traitSet    The traits of this rel.
+     * @param input     Input relational expression.
+     * @param all   Whether this operator should return all rows or only distinct rows.
+     * @param rowType  Row type this expression produces.
      */
     public IgniteReduceIntersect(
             RelOptCluster cluster,
             RelTraitSet traitSet,
             RelNode input,
             boolean all,
-            RelDataType rowType
+            RelDataType rowType,
```

Comment:
It looks like we don't need to pass rowType from outside, as it should be constructed at first call 
`getRowType() -> deriveRowType() -> buildRowType()`
I can't understand, why we initialize protected field below `this.rowType = rowType`, that prevents building the rowType and therefore makes buildRowType() method totally useless....

Seems, the original rowType was passed just for correct `aggregateFieldsCount()`...